app/scripts/lib/security-provider-helpers.ts

```diff
@@ -1,35 +1,46 @@
+import { Json } from '@metamask/utils';
+import { MessageParams } from '@metamask/message-manager';
 import getFetchWithTimeout from '../../../shared/modules/fetch-with-timeout';
 import { MESSAGE_TYPE } from '../../../shared/constants/app';
 
 const fetchWithTimeout = getFetchWithTimeout();
 
+export interface RequestData {
+  origin: string;
+  msgParams?: MessageParams;
+  messageParams?: MessageParams;
+  txParams?: Record<string, unknown>;
+}
+
 export async function securityProviderCheck(
-  requestData,
-  methodName,
-  chainId,
-  currentLocale,
-) {
+  requestData: RequestData,
```

Comment:
Following the above comment about types, we could define a `TransactionRequestData` and `MessageRequestData` and use a union here to associate them with the possible method names?